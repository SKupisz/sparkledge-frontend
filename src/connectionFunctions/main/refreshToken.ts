/* 
    The refreshToken function refreshes user's token
*/

import axios from "axios";

const refreshToken = async (
  userRefreshToken: string, 
  changeAccessToken: (newToken: string) => void,
  changeRefreshToken: (newToken: string) => void,
) => {
  await axios.post(`${process.env.REACT_APP_CONNECTION_TO_SERVER}/users/refresh`, {}, {
    headers: {
      Authorization: `Bearer ${userRefreshToken}`,
    },
  })
    .then((res) => {
      changeAccessToken(res.data.accessToken);
      changeRefreshToken(res.data.refreshToken);
    })
    .catch((err) => {
      console.log(err);
    });
};

export default refreshToken;
