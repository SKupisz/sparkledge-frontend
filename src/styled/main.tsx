import styled from "styled-components";

export const LightMode = {
    navBgColor: "#eeeeee",
    navElemBgColor: "rgba(240,240,240,.2)",
    bgColor: "#f0f0f0",
    filterColor: "rgba(0,0,0,.2)",
    color: "#0A77BF",
    landingButtonColor: "rgba(240,240,240,.5)",
    fonts: {
        main: "Mohave, sans-serif",
        textShadowMain: "3px 3px 4px rgba(0,0,0,.05)"
    }
}

export const DarkMode = {
    navBgColor: "#0c0c0c",
    navElemBgColor: "rgba(20,20,20,.2)",
    bgColor: "#111",
    filterColor: "rgba(0,0,0,.7)",
    color: "#0895E6",
    landingButtonColor: "rgba(20,20,20,.5)",
    fonts: {
        main: "Mohave, sans-serif",
        textShadowMain: "3px 3px 4px rgba(0,0,0,.05)"
    }
}

export const MainContainer = styled.section`
    width: 100%;
    height: 90vh;
    position: relative;
    top: 10vh;
    background: ${(props) => props.theme.bgColor};

    a{
        width: fit-content !important;
    }
`;