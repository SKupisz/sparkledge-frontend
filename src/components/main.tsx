import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import { useSelector } from "react-redux";

import { LightMode, DarkMode, SparkledgeGlobalStyle } from "../styled/main";
import { RootState } from "../redux/mainReducer";
import Navbar from "./subcomponents/navbar";
import Welcome from "./subpages/welcome";
import About from "./subpages/about";
import SigningPanel from "./subpages/signing";
import Searcher from "./subpages/searcher";
import Notfound from "./subpages/notfound";

const Main: React.FC = () => {
  const graphicalMode: number = useSelector(
    (state: RootState) => state.generalData.graphicalMode
  );

  return (
    <ThemeProvider theme={graphicalMode === 0 ? LightMode : DarkMode}>
      <SparkledgeGlobalStyle isLight={graphicalMode === 0 ? true : false}/>
      <Router>
        <Navbar />
        <Routes>
          <Route path="/" element={<Welcome />} />
          <Route path="/about" element={<About />} />
          <Route path="/signin" element={<SigningPanel mode={1} />} />
          <Route path="/searcher" element={<Searcher/>}/>
          <Route path="*" element={<Notfound />} />
        </Routes>
      </Router>
    </ThemeProvider>
  );
};

export default Main;
