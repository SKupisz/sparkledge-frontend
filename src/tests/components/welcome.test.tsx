import React from "react";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import { ThemeProvider } from "styled-components";
import Welcome from "../../components/subpages/welcome";
import LandingComponent from "../../components/helperComponents/welcome/landingComponent";
import DescribeComponent from "../../components/helperComponents/welcome/describeComponents";
import DescribeComponentWidget from "../../components/helperComponents/welcome/describeComponentWidget";
import FooterComponent from "../../components/helperComponents/welcome/footerComponent";
import { DarkMode } from "../../styled/main";

describe("Welcome site rendering tests", () => {
  it("testing if Welcome is rendering", () => {
    expect.assertions(1);
    expect(toJson(shallow(<Welcome />))).toMatchSnapshot();
  });

  it("testing if LandingComponent is rendering", () => {
    expect(toJson(shallow(<LandingComponent />))).toMatchSnapshot();
  });

  it("testing if DescribeComponent is rendering", () => {
    expect(toJson(shallow(<DescribeComponent />))).toMatchSnapshot();
  });
    
  it("testing if FooterComponent is rendering", () => {
    expect(toJson(shallow(<FooterComponent />))).toMatchSnapshot();
  });

  it("testing is DescribeComponentWidget is rendering", () => {
    const wrapper = mount(<ThemeProvider theme={DarkMode}>
      <DescribeComponentWidget header="test header'a" content="test" />
    </ThemeProvider>);

    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find("header").text()).toBe("test header'a");
  });
});
