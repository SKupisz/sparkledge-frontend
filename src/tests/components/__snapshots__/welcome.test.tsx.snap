// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome site rendering tests testing if DescribeComponent is rendering 1`] = `
<styled.section
  backgroundRepeat="repeat"
  backgroundSize="initial"
  bottomPadding={10}
  className="block-center"
  reversedShadow={true}
  source="pattern_background4_1.webp"
>
  <styled.section>
    <styled.header
      className="block-center"
    >
      Co może Tobie dać korzystanie ze Sparkledge?
    </styled.header>
    <styled.section
      className="block-center"
    >
      <DescribeComponentWidget
        header="Dostęp do wiedzy"
      >
        Na naszej platformie znajdziesz notataki pogrupowane na przedmioty i kierunki studiów. Sa to notataki innych studentów, dzięki temu wiesz ze sa one dokładnie tym czego wymagają wykładowcy. 
      </DescribeComponentWidget>
      <DescribeComponentWidget
        header="Kontakt ze społecznością akademicką"
      >
        Wraz ze znajomymi za pomoca naszej platformy mozecie wymieniać się notataki oraz trzymać je w jednym miejscu. Twoi znajomi oceniają je, a dzięki ich ocenom ty masz dostęp do większej ilości notatek innych użytkowników 
      </DescribeComponentWidget>
      <DescribeComponentWidget
        header="Ciekawe benefity na uczelni"
      >
        Dzięki korzystaniu z naszej platformy nie tracisz czas na nauke rzeczy których i tak nikt od ciebie nie będzie wymagał. Są to notatki studentów którzy już zdali ten przedmiot albo sa w trakcie i notują na wykladach. Znajdziesz tutaj dokładnie to czego potrzeba aby zdać.
      </DescribeComponentWidget>
    </styled.section>
    <styled.div
      className="block-center"
      marginBottom={4}
    >
      <Link
        to="/searcher"
      >
        <styled.div
          className="block-center"
        >
          Sprawdź wyszukiwarkę
        </styled.div>
      </Link>
    </styled.div>
  </styled.section>
</styled.section>
`;

exports[`Welcome site rendering tests testing if FooterComponent is rendering 1`] = `
<styled.footer
  className="block-center"
>
  <styled.section
    className="block-center"
  >
    <styled.div>
      <Link
        key="column-0-elem"
        to="/"
      >
        <styled.div>
          <styled.img
            src="sparkledge_logo.webp"
          />
        </styled.div>
      </Link>
    </styled.div>
    <styled.div>
      <Link
        key="column-1-elem"
        to="/"
      >
        <styled.div>
          Główna
        </styled.div>
      </Link>
      <Link
        key="column-1-elem"
        to="/searcher"
      >
        <styled.div>
          Wyszukiwarka
        </styled.div>
      </Link>
      <Link
        key="column-1-elem"
        to="/signin"
      >
        <styled.div>
          Zaloguj się
        </styled.div>
      </Link>
      <Link
        key="column-1-elem"
        to="/signup"
      >
        <styled.div>
          Zarejestruj się
        </styled.div>
      </Link>
    </styled.div>
    <styled.div>
      <Link
        key="column-2-elem"
        to="/about"
      >
        <styled.div>
          O nas
        </styled.div>
      </Link>
      <a
        href="https://www.vecteezy.com/free-vector/education"
        key="column-2-elem"
        rel="noreferrer"
        target="_blank"
      >
        <styled.div>
          Education Vectors by Vecteezy
        </styled.div>
      </a>
    </styled.div>
  </styled.section>
</styled.footer>
`;

exports[`Welcome site rendering tests testing if LandingComponent is rendering 1`] = `
<styled.section
  className="block-center"
  source="academic_background_edited_2.webp"
>
  <styled.section>
    <styled.header
      className="block-center"
      height={0}
    >
      <span
        className="block-center"
      >
        Sparkledge
      </span>
      <styled.span>
        Spark of your knowledge
      </styled.span>
    </styled.header>
    <styled.div
      className="block-center"
      leftPos={100}
    >
      Sparkledge to platforma łącząca studentów, dającą im możliwość rozwoju i osiągania lepszych wyników. Sparklege to platforma stworzona przez studentów dla studentów. Znajdziesz na niej notatki Twoich znajomych oraz absolwentów ocenione przez innych użytkowników. Dzięki niej nauka do kolosów będzie przyjemniejsza. Jedyne co musisz zrobić aby dostać dostęp do skarbnicy wiedzy to załączyć swoje notataki
    </styled.div>
    <styled.div
      className="block-center"
      rightPos={100}
    >
      <Link
        to="/signup"
      >
        <styled.div
          className="block-center"
        >
          Dołącz do nas
        </styled.div>
      </Link>
    </styled.div>
  </styled.section>
</styled.section>
`;

exports[`Welcome site rendering tests testing if Welcome is rendering 1`] = `
<styled.section
  className="block-center"
>
  <Suspense
    fallback={
      <styled.div
        className="block-center"
      >
        Ładowanie...
      </styled.div>
    }
  >
    <lazy />
    <lazy />
    <lazy />
  </Suspense>
</styled.section>
`;

exports[`Welcome site rendering tests testing is DescribeComponentWidget is rendering 1`] = `
<Component
  theme={
    Object {
      "bgColor": "#111",
      "color": "#0895E6",
      "errorColor": "rgba(240,30,30,.75)",
      "filterColor": "rgba(0,0,0,.7)",
      "filtersButtonBackground": "rgba(3,73,119,.9)",
      "filtersButtonResponsiveBackground": "rgba(10,10,10,.9)",
      "filtersResponsiveBackground": "rgba(40,40,40,.9)",
      "fonts": Object {
        "main": "Mohave, sans-serif",
        "textShadowMain": "3px 3px 4px rgba(0,0,0,.05)",
      },
      "landingButtonColor": "rgba(20,20,20,.5)",
      "navBgColor": "#0c0c0c",
      "navElemBgColor": "rgba(20,20,20,.2)",
      "resultBackground": "rgba(24,87,255,.15)",
      "signingInputBackground": "rgba(40,40,40,.7)",
    }
  }
>
  <DescribeComponentWidget
    content="test"
    header="test header'a"
  >
    <styled.div>
      <div
        className="sc-iqseJM dbncUd"
      >
        <styled.header
          className="block-center"
        >
          <header
            className="sc-crHmcD eabHxR block-center"
          >
            test header'a
          </header>
        </styled.header>
        <styled.div
          className="block-center"
        >
          <div
            className="sc-egiyK faVmmf block-center"
          />
        </styled.div>
      </div>
    </styled.div>
  </DescribeComponentWidget>
</Component>
`;
